class-pinterest-for-woocommerce.php

```diff
@@ -247,6 +247,9 @@ public function init_plugin() {
 
 			// Disconnect advertiser if advertiser or tag change.
 			add_action( 'update_option_pinterest_for_woocommerce', array( $this, 'maybe_disconnect_advertiser' ), 10, 2 );
+
+			// Unschedule tasks if woocommerce is deactivated.
+			add_action( 'deactivate_woocommerce/woocommerce.php', array( Pinterest\ProductSync::class, 'cancel_jobs' ) );
```

Comment:
There is an edge case that I think we should consider here: sometimes the WooCommerce plugin can be uploaded to a site with a main directory named something other than `woocommerce`. In that scenario, this action will be missed. 

I think we should utilize the [`WC_PLUGIN_FILE`](https://github.com/woocommerce/woocommerce/blob/aa8b7a4a6b7d21ccddaac360766c4daa396b9a98/plugins/woocommerce/woocommerce.php#L19-L21) constant to help here. This method we're in is run on the `plugins_loaded` hook, so that means that WooCommerce will have been loaded by this point (if it is active).

```suggestion
			if ( defined( 'WC_PLUGIN_FILE' ) ) {
				register_deactivation_hook( WC_PLUGIN_FILE, array( Pinterest\ProductSync::class, 'cancel_jobs' ) );
			}
```

I think that this approach will overcome this edge case, as the `WC_PLUGIN_FILE` constant will be defined based on the actual path, regardless of any directory naming differences.